sklearn/linear_model/_stochastic_gradient.py

```diff
@@ -1087,10 +1097,18 @@ class SGDClassifier(BaseSGDClassifier):
           Each time n_iter_no_change consecutive epochs fail to decrease the
           training loss by tol or fail to increase validation score by tol if
           `early_stopping` is `True`, the current learning rate is divided by 5.
+        - 'pa1': passive-aggressive algorithm 1, see [1]_. Only with `loss='hinge'`.
+          Update is `w += eta y x` with `eta = min(PA_C, loss/||x||**2)`.
+        - 'pa2': passive-aggressive algorithm 2, see [1]_. Only with
+          `loss='squared_hinge'`.
```

Comment:
It looks like we raise an error in the base class if the loss is not `hinge`. If we want to allow `squared_hinge` for `pa2` then maybe the validation check should be updated.

Reply:
That is by intention. The analogy to squared hinge has its appeal (not mentioned in the paper, only in scikit-learn's PR), but in effect the used loss is always hinge, never squared hinge.

Reply:
Got it. I looked at the current code for the Passive Aggressive classes and it's clear.

Reply:
The docstring for pa2 here was incorrect. I pushed a fix.